src/libraries/System.Collections.Immutable/src/System/Collections/Immutable/IImmutableDictionary.cs

```diff
@@ -15,7 +15,7 @@ namespace System.Collections.Immutable
     /// <typeparam name="TValue">The type of the value.</typeparam>
     [SuppressMessage("Microsoft.Naming", "CA1711:IdentifiersShouldNotHaveIncorrectSuffix")]
     [SuppressMessage("Microsoft.Naming", "CA1710:IdentifiersShouldHaveCorrectSuffix", Justification = "Ignored")]
-    public interface IImmutableDictionary<TKey, TValue> : IReadOnlyDictionary<TKey, TValue>
+    public interface IImmutableDictionary<TKey, TValue> : IReadOnlyDictionary<TKey, TValue> where TKey : notnull
```

Comment:
@stephentoub I think this should also follow `IDictionary` and `IReadOnlyDictionary` to don't constraint `TKey` to `notnull`. However the docs don't state it that way, but since this is the immutable equivalent of that interface I would like to be consistent.

Reply:
@AArnott, @sharwell, what was the intention for the `IImutableXx` interfaces.  Are they intended to be implemented by anything other than the corresponding `ImmutableXx` types?  Should they allow null TKeys?

Reply:
I was wondering about that, was planning to pull those change after his PR merged.
Some type of dictionaries require not null key, assume we can keep notnull constraint for them

Reply:
The interfaces are certainly allowed to be implemented elsewhere. We developed them because in fact there are alternative implementations of these classes with different perf characteristics that folks may want to use and still interop with others who are willing to just take the interface.

I never realized there were dictionaries that accepted null keys. If that's the case, maybe we should leave the constraint off. But I personally wouldn't find it disruptive to anything I do to have that constraint present. I would do whatever `IReadOnlyDictionary` does here for consistency.

Reply:
Ok, thanks.  We're removing the `TKey : notnull` from `IDictionary<TKey, TValue>` and `IReadOnlyDictionary<TKey, TValue>`... sounds like we should remove it from here as well.

Reply:
Ok, will remove, thank you